File: src/IO_24/_02_Бондаренко_Тарас_Андрійович/lab5/Punctuation.java
package IO_24._02_Бондаренко_Тарас_Андрійович.lab5;

public class Punctuation implements SentenceElement {
    private final String character;

    public Punctuation(String character) {
        this.character = character;
    }

    @Override
    public String toString() {
        return character;
    }
}

File: Java/lab_5/Punctuation.kt
package OOP.Java.lab_5

/**
 * A class representing a punctuation mark following a word or a sentence.
 *
 * @property punctuationMark a [String] value of a punctuation mark.
 */
class Punctuation(var punctuationMark: String) {

    init {
        if (!"\\p{Punct}".toRegex().containsMatchIn(punctuationMark)) {
            this.punctuationMark = ""
        }
    }

    override fun toString(): String {
        return punctuationMark
    }
}